### PR TITLE
Refactor Redis commands and connection pool

### PR DESCRIPTION
**Pull request by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg**
Simpler version of https://github.com/envoyproxy/envoy/pull/6055. Will amend description/close other PR if this one passes coverage test without a strange bison/flex error.

### CODE REVIEW

**Comment by mattklein123** [comment]
@FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg your actual error is:
```Code coverage 97.3 is lower than limit of 97.5```, somehow you lowered the coverage to below the threshold.

**Comment by HenryYYang** [comment]
+1

**Comment by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg** [comment]
@mattklein123 don't know how I missed that in the bottom of the output there. In any case, it is fixed.

File: source/extensions/filters/network/common/redis/BUILD [hunk 1]
```diff
@@ -0,0 +1,40 @@
+licenses(["notice"])  # Apache 2
+
+# Redis proxy L4 network filter. Implements consistent hashing and observability for large redis
```
Comment (mattklein123) [inline source/extensions/filters/network/common/redis/BUILD]: Can you delete this comment as I don't think applies here now that it's common code?

File: test/extensions/filters/network/common/redis/BUILD [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+load(
+    "//bazel:envoy_build_system.bzl",
+    "envoy_cc_mock",
+    "envoy_package",
+)
+load(
+    "//test/extensions:extensions_build_system.bzl",
+    "envoy_cc_test",
+    "envoy_extension_cc_test_binary",
```
Comment (mattklein123) [inline test/extensions/filters/network/common/redis/BUILD]: This is not used, I would remove this entire load() and just import the normal envoy_cc_test from the normal build size bzl file.

File: test/extensions/filters/network/common/redis/codec_impl_test.cc [hunk 3]
```diff
@@ -65,6 +66,18 @@ TEST_F(RedisEncoderDecoderImplTest, SimpleString) {
   EXPECT_EQ(0UL, buffer_.length());
 }
 
+TEST_F(RedisEncoderDecoderImplTest, BulkString) {
```
Comment (mattklein123) [inline test/extensions/filters/network/common/redis/codec_impl_test.cc]: Was this test missing? Why was this added?

File: source/extensions/filters/network/common/redis/BUILD (hunk 1 above)
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply source/extensions/filters/network/common/redis/BUILD]: done

File: test/extensions/filters/network/common/redis/BUILD (hunk 2 above)
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply test/extensions/filters/network/common/redis/BUILD]: done

File: test/extensions/filters/network/common/redis/codec_impl_test.cc (hunk 3 above)
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply test/extensions/filters/network/common/redis/codec_impl_test.cc]: This test is new- since a bunch of the other types are there, I figured I'd add it. In theory the array tests capture it; I think one could probably remove some of the other types and do them in the array tests too, but it reads nicer. I can remove if you prefer.

Comment (mattklein123) [reply test/extensions/filters/network/common/redis/codec_impl_test.cc]: Nope thanks for adding, just curious.
